lib/active_record/virtual_attributes/virtual_delegates.rb

```diff
@@ -46,38 +46,21 @@ def virtual_delegate(*methods)
               options[:to] = to
             end
 
+            unless (to_ref = reflection_with_virtual(to.to_s))
+              raise ArgumentError, "Delegation needs an association. Association #{to} does not exist"
```

Comment:
Similar question here as as the type... should we deprecate on the current release and do this raise in the new version?

Reply:
+1 for dropping it and making it explicit.  If in the future something comes along where we want it dynamic we can add it back in later, hopefully better.

Reply:
Sorry I missed your question Joe.
The `to_ref` has always been required. This just moved the check from one method to another.